src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -423,6 +423,11 @@ private static SyntaxNode GetLeftmostCondition(SyntaxNode node)
                         continue;
                     case SyntaxKind.ReturnStatement:
                         node = ((ReturnStatementSyntax)node).Expression;
+                        if (node == null)
+                        {
+                            return null;
+                        }
```

Comment:
consider adding the null check above the switch-statement.  then we can remove all the null checks inside the switch cases.